TicTacToe. Добавила ДЗ № 2 "Крестики-нолики"

fedyaeva opened the pull request:
Добавила ДЗ № 2 "Крестики-нолики"

fedyaeva committed:
Добавила ДЗ № 2 "Крестики-нолики"

fedyaeva committed:
Update Program.cs

fedyaeva committed:
Revert "Update Program.cs"

This reverts commit 4825b9dbec7ebb06d96c3ec38b11d7bf18bb68aa.

fedyaeva committed:
TicTacToe. Поправила ошибку с выводом сообщения о ничье при выигрыше на последнем ходе

TicTacToe. Поправила ошибку с выводом сообщения о ничье при выигрыше на последнем ходе

Dezodemius commented on TicTacToe/TicTacToe/Program.cs:
```diff
@@ -0,0 +1,127 @@
+{
+    static void Main(string[] args)
+    {
+        string[,] field = {
```
Давай весь код вынесем в отдельный класс, к которому потом здесь обратимся и запустим игру

Dezodemius commented on TicTacToe/TicTacToe/Program.cs:
```diff
@@ -0,0 +1,127 @@
+        Move();
+        DrawField(field);
+       }
+        void Move(){
```
Метод объявлен внутри метода Main. Так не стоит делать. Его объявление должно быть за пределами других методов

Dezodemius replied:
Нужно убедиться, что каждый метод находится в пределах класса, а не внутри другого метода

Dezodemius commented on TicTacToe/TicTacToe/Program.cs:
```diff
@@ -0,0 +1,127 @@
+        void Move(){
+        //Ввод ячейки
+        int number;
+        while (true)
```
Вот у тебя выше же есть хорошая практика как правильно реализовывать такой цикл. Давай сделаем что-то похожее и здесь

Dezodemius commented on TicTacToe/TicTacToe/Program.cs:
```diff
@@ -0,0 +1,127 @@
+            string enteredValue = Console.ReadLine();
+            if (int.TryParse(enteredValue, out number))
+            {
+                if (number >= 1 && number <= 9)
```
1 и 9 какие-то магические числа. Давай их вынесем в переменные с говорящими названиями

Dezodemius commented on TicTacToe/TicTacToe/Program.cs:
```diff
@@ -0,0 +1,127 @@
+            {
+                if (number >= 1 && number <= 9)
+                    break;
+            } Console.WriteLine("Номер ячейки указан некорректно");
```
Кажется, сообщение не очень говорящее. "Некорректно" может отражать разные проблемы, каждая из которых имеет своё решение

Dezodemius commented on TicTacToe/TicTacToe/Program.cs:
```diff
@@ -0,0 +1,127 @@
+        }
+        string searchCell = $"{number}";
+        
+        //Поиск ячейки и выполнение хода
```
Все комментарии давай удалим. Лучше использовать документацию `///<summary>`. Я на лекциях показывал

Dezodemius commented on TicTacToe/TicTacToe/Program.cs:
```diff
@@ -0,0 +1,127 @@
+                        Console.WriteLine($"Ход № {moveCount} выполнен.");
+                        CheckWin();
+                        CheckDrow();
+                        currentPlayer = (currentPlayer == "X") ? "O" : "X";
```
"O" и "X" много где используются. Давай их вынесем в константы класса

fedyaeva committed:
Update Program.cs

fedyaeva committed:
Revert "Update Program.cs"

This reverts commit f5956104d0082bdc4a5bbcaaaa2fbd0b6a846040.

fedyaeva committed:
Исправление замечаний

Исправила замечания:
1. Вынесла весь код в отдельный класс
2. Исправила объявление методов внутри другого метода
3. Исправила условие выполнения цикла while в методе Move()
4. Вынесла в переменные номера первой и последней ячейки поля
5. Исправила сообщение о неправильном указании номера ячейки в методе Move()
6. Добавила документацию к каждому методу
7. Вынесла "O" и "X" в константы класса
